source/Preprocessor.cpp

```diff
@@ -417,6 +418,23 @@ Trivia Preprocessor::handleTimescaleDirective(Token directive) {
     return Trivia(TriviaKind::Directive, timescale);
 }
 
+Trivia Preprocessor::handleLineDirective(Token directive) {
+    Token lineNumber = expect(TokenKind::IntegerLiteral);
+    Token fileName = expect(TokenKind::StringLiteral);
+    Token level = expect(TokenKind::IntegerLiteral);
+    const SVInt& levNum = std::get<SVInt>(level.numericValue());
+    if (!(levNum == 0 || levNum == 1 || levNum == 2)) {
```

Comment:
If any of lineNumber, fileName, or level are missing we should probably return instead of calling addLineDirective (we should still create the LineDirectiveSyntax though). The expect() call will generate a token with the Missing flag set if it couldn't find the thing we wanted.